tools/walletextension/httpapi/routes.go

```diff
@@ -517,3 +523,66 @@ func getMessageRequestHandler(walletExt *services.Services, conn UserConn) {
 		walletExt.Logger().Error("error writing success response", log.ErrKey, err)
 	}
 }
+
+func keyExchangeRequestHandler(walletExt *services.Services, conn UserConn) {
+	// read the request
+	body, err := conn.ReadRequest()
+	if err != nil {
+		handleError(conn, walletExt.Logger(), fmt.Errorf("error reading request: %w", err))
+		return
+	}
+
+	// Deserialize the attestation report
+	var attestationReport tencommon.AttestationReport
+	if err := json.Unmarshal(body, &attestationReport); err != nil {
+		handleError(conn, walletExt.Logger(), fmt.Errorf("error unmarshaling attestation report: %w", err))
+		return
+	}
+
+	// Verify the attestation report
+	verifiedData, err := keymanager.VerifyReport(&attestationReport)
+	if err != nil {
+		handleError(conn, walletExt.Logger(), fmt.Errorf("error verifying attestation report: %w", err))
+		return
+	}
+
+	// TODO: How do we need to handle the verified data? What should we check to be sure it is correct?
+	// Is this the hash of the public key?
```

Comment:
you put the public key directly in the report I think? So you don't need to hash, just check the `verifiedData == report.PubKey`.

In the node enclave we have a json obj (pubkey, host address and stuff) and we put the hash of that into the report. So on the verify side we have to reproduce that and check if `hash(jsonData) == verifiedData`.

Reply:
I think we need hash, otherwise we get the following error:
`REQ: Failed to get attestation report    component=wallet_extension error="reportData too large"`